HW0/P4/P4.py

```diff
@@ -29,49 +29,118 @@
     #
     # Load true trajectory and plot it
     # Normally, this data wouldn't be available in the real world
+
+read_temp=np.loadtxt('P4_trajectory.txt',delimiter=',')
+xx=(np.transpose(read_temp))[0]
+yy=(np.transpose(read_temp))[1]
+zz=(np.transpose(read_temp))[2]
     #####################
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '--b', label='True trajectory')
+ax.plot(xx, yy, zz,'--b', label='True trajectory')
 
     #####################
     # Part 2:
     #
     # Read the observation array and plot it (Part 2)
-    #####################
+read_temp_meas=np.loadtxt('P4_measurements.txt',delimiter=',')
+C=np.matrix([[1/rx,0,0],[0,1/ry,0],[0,0,1/rz]])
+sm=np.transpose(read_temp_meas)
+xk=C*sm
+x_apx=np.squeeze(np.asarray(xk[0]))
+y_apx=np.squeeze(np.asarray(xk[1]))
+z_apx=np.squeeze(np.asarray(xk[2])) 
+	#####################
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+ax.plot(x_apx, y_apx, z_apx,'.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
-
-    # A = ?
-    # a = ?
+s0=np.matrix([[0],[0],[2],[15],[3.5],[4.0]])
+A=np.matrix([[1,0,0,dt,0,0],[0,1,0,0,dt,0],[0,0,1,0,0,dt],[0,0,0,1-c*dt,0,0],[0,0,0,0,1-c*dt,0],[0,0,0,0,0,1-c*dt]])
+a=np.matrix([[0],[0],[0],[0],[0],[g*dt]])
+s=np.zeros((120,6))
+    
     # s = ?
 
-    # Initial conditions for s0
-    # Compute the rest of sk using Eq (1)
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '-k', label='Blind trajectory')
+    # calculate each position
+s_temp=s0
+s[0]=np.transpose(s0)
+i=1
+while i < 120:
+	s_temp1=A*s_temp+a
+	s[i]=np.transpose(s_temp1)
+	s_temp=s_temp1
+	i+=1
+
 
```

Comment:
I have two recommendations:
1- Instead of using 120 it is better to use for example len(read_temp_meas), in this way your code will be more robust.
2- Instead of using a "while" condition, you should make a "for" loop.  "while" is usually used when a command must be executed until a defined condition is met. "For" is used for repeating the same action a defined number of times. For example:

for i in xrange(120):
        s_temp1=A*s_temp+a
	s[i]=np.transpose(s_temp1)
	s_temp=s_temp1